scripts/install.sh

```diff
@@ -0,0 +1,187 @@
+fi
+tty_mkbold() { tty_escape "1;$1"; }
+tty_underline="$(tty_escape "4;39")"
+tty_blue="$(tty_mkbold 34)"
+tty_red="$(tty_mkbold 31)"
+tty_bold="$(tty_mkbold 39)"
+tty_reset="$(tty_escape 0)"
+
+ohai() {
+  printf "${tty_blue}==>${tty_bold} %s${tty_reset}\n" "$(shell_join "$@")"
+}
+
+linux_install_pre() {
+    sudo apt-get update 
+    sudo apt-get install --no-install-recommends --no-install-suggests -y apt-utils curl git cmake build-essential unzip 
+}
+
+linux_install_python() {
+    which -s python3.7
+    if [[ $? != 0 ]] ; then
+        ohai "Installing python3.7"
+        sudo apt-get install --no-install-recommends --no-install-suggests -y python3.7 python-pip python3-dev   
+    else
+        ohai "Updating python3.7"
+        sudo apt-get update python3.7
+    fi
+    ohai "Installing bittensor deps"
+    python3.7 -m pip install python-dev
+    python3.7 -m pip install --upgrade pip
+}
+
+mac_install_xcode() {
+    which -s xcode-select
+    if [[ $? != 0 ]] ; then
+        ohai "Installing xcode:"
+        xcode-select --install
+    fi
+}
+
+mac_install_brew() {
+    which -s brew
+    if [[ $? != 0 ]] ; then
+        ohai "Installing brew:"
+        ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+    else
+        ohai "Updating brew:"
+        brew update
+    fi
+}
+
+mac_install_cmake() {
+    which -s cmake
+    if [[ $? != 0 ]] ; then
+        ohai "Installing cmake:"
+        brew install cmake
+    else
+        ohai "Updating cmake:"
+        brew upgrade cmake
+    fi
+}
+
+mac_install_python() {
+    which -s python3.7
+    if [[ $? != 0 ]] ; then
+        ohai "Installing python3.7"
+        brew install python@3.7
+    else
+        ohai "Updating python3.7"
+        brew upgrade python@3.7
+    fi
+    ohai "Installing bittensor deps"
+    python3.7 -m pip install python-dev
+    python3.7 -m pip install --upgrade pip
+}
+
+install_bittensor() {
+    mkdir tmp
+    cd tmp
+    ohai "Cloning bittensor"
+    git clone https://github.com/opentensor/bittensor.git
+    cd bittensor
+    ohai "Installing bittensor"
+    python3.7 -m pip install -e .
+    cd ../..
+    rm -rf tmp
+}
+
+# Do install.
+OS="$(uname)"
+if [[ "$OS" == "Linux" ]]; then
+    ohai "This script will install:"
+    echo "git"
+    echo "curl"
+    echo "git"
+    echo "cmake"
+    echo "build-essential"
+    echo "python3.7"
+    echo "python3.7-pip"
+
+    wait_for_user
+    linux_install_pre
+    linux_install_python
+    install_bittensor
+    ohai "Installation successful!"
+
+elif [[ "$OS" == "Darwin" ]]; then
+    ohai "This script will install:"
+    echo "xcode"
+    echo "homebrew"
+    echo "git"
+    echo "cmake"
+    echo "python3.7"
+    echo "python3.7-pip"
+
+    wait_for_user
+    mac_install_brew
+    mac_install_cmake
+    mac_install_python
+    install_bittensor
+    ohai "Installation successful!"
+    
+
+else
+  abort "Bittensor is only supported on macOS and Linux."
```

Comment:
Change Linux to Ubuntu (version ... and up). The aptitude package manager is not available in other distros.
Arch linux for instance has pacman, Centos has yum, etc ...

Reply:
Very good point

Reply:
Stackoverflow say's this might work

if [  -n "$(uname -a | grep Ubuntu)" ]; then
    sudo apt-get update && sudo apt-get upgrade 
else
    yum update
fi  

perhaps I should try catch for apt?
which -s apt 
for instance?